packages/client-assertion-validation/src/utils.ts

```diff
@@ -123,18 +124,27 @@ export const validateKid = (kid?: string): ValidationResult<string> => {
 
 export const validateAudience = (
   aud: string | string[] | undefined
-): ValidationResult<string[]> => {
-  if (aud === config.clientAssertionAudience) {
-    return successfulValidation([aud]);
+): ValidationResult<string[] | string> => {
+  if (!aud) {
+    return failedValidation([audienceNotFound()]);
   }
 
-  if (!Array.isArray(aud)) {
-    return failedValidation([invalidAudienceFormat()]);
-  }
-  if (!aud.includes(config.clientAssertionAudience)) {
+  if (Array.isArray(aud)) {
+    if (config.clientAssertionAudience.every((entry) => aud.includes(entry))) {
+      return successfulValidation(aud);
+    }
+    return failedValidation([invalidAudience()]);
+  } else {
+    const split = aud.split(",").map((s) => s.trim());
+    if (split.length > 1) {
+      return failedValidation([invalidAudienceFormat()]);
+    }
```

Comment:
This check is not strictly necessary.
If we treat the `aud` as a single string and it contains more characters than expected, then the next check will block the validation.

Reply:
In this way we raise a custom error when the input is a comma-separated string with multiple values.